google_health/delphi_google_health/run.py

```diff
@@ -37,32 +37,27 @@ def run_module():
     start_time = time.time()
     csv_export_count = 0
     oldest_final_export_date = None
-    
+
     # read parameters
     params = read_params()
-    ght_key = params["ght_key"]
-    start_date = params["start_date"]
-    end_date = params["end_date"]
-    static_dir = params["static_file_dir"]
-    export_dir = params["export_dir"]
-    data_dir = params["data_dir"]
-    wip_signal = params["wip_signal"]
-    cache_dir = params["cache_dir"]
+    ght_key = params["indicator"]["ght_key"]
+    start_date = params["indicator"]["start_date"]
+    end_date = params["indicator"]["end_date"]
+    static_dir = params["indicator"]["static_file_dir"]
+    export_dir = params["common"]["export_dir"]
+    data_dir = params["indicator"]["data_dir"]
+    wip_signal = params["indicator"]["wip_signal"]
 
     logger = get_structured_logger(
-        __name__, filename=params.get("log_filename"),
-        log_exceptions=params.get("log_exceptions", True))
-
-    arch_diff = S3ArchiveDiffer(
-        cache_dir, export_dir,
-        params["bucket_name"], "ght",
-        params["aws_credentials"])
-    arch_diff.update_cache()
-    print(arch_diff)
-    # if missing start_date, set to today (GMT) minus 5 days
-    if start_date == "":
-        now = datetime.datetime.now(datetime.timezone.utc)
-        start_date = (now - datetime.timedelta(days=4)).strftime("%Y-%m-%d")
+        __name__, filename=params["common"].get("log_filename"),
+        log_exceptions=params["common"].get("log_exceptions", True))
+
+    if params["archive"]:
```

Comment:
if there's any chance archive isn't in the json instead of having an empty value, should use `.get()` here

Reply:
Not sure if this is important to your thinking on this, but the point here is to use the existence of the `archive` field as a marker of whether to do archiving at all rather than as a default value safeguard.

Reply:
with the current approach, if no archiving is desired, you still need an `archive: {}` (or None or some other falsy value) and can't remove the archive key entirely from the json. If that's the intention that sgtm, but wasn't sure.

Reply:
why can't you remove it entirely?

Reply:
won't it keyerror?

Reply:
yes of course.  Fixed to do what I meant